src/OmniSharp.MSBuild/ProjectFile/ProjectFileInfo.ProjectData.cs

```diff
@@ -42,6 +43,7 @@ private class ProjectData
             public ImmutableArray<string> References { get; }
             public ImmutableArray<PackageReference> PackageReferences { get; }
             public ImmutableArray<string> Analyzers { get; }
+            public RuleSet RuleSet { get; }
```

Comment:
the initial implementation might not need it but you probably want to support editorconfig as well since that can change some of built-in roslyn analyzers behavior if you want to have the same experience as in VS.

we are working on to make it automatically picked up by Roslyn, so you could just wait until then for editorconfig work (so that msbuild has editorconfig info in it like Analyzers) and you can build roslyn solution with that info like what you are doing with AnalyzerFileReference and CompilationOptions. and then after that CompilationWithAnalyzer takes care of all the detail once context is setup right.

until then, if you want to support editorconfig, you have to do it yourself. let me know if you want to, it is a bit of work depends on what infrastructure you have on editorconfig.

Reply:
in simple term, you need to set up http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures.Wpf/Options/EditorConfigDocumentOptionsProvider.cs,19 in your workspace.

Reply:
@heejaechang do you know when roslyn integrated solution will be released? Not exact timing but like is it upcoming month or maybe year away?

Reply:
@savpek can't say for sure. but we are actively working on. we recently checked in changes in compiler layer. and I believe we are working on to make it work on roslyn workspace layer. and then features and etc.